File: app/src/main/java/co/instameals/dinnersolutions/api/v2/utils/Relation.java
package co.instameals.dinnersolutions.api.v2.utils;

import java.util.UUID;

public class Relation<T> {
    // FIXME: make this optional; create a JSON parser for it
    public T relatedObject;
    private final UUID relationId;

    public Relation(UUID relationId) {
        this.relationId = relationId;
    }
}

File: app/src/main/java/co/instameals/dinnersolutions/api/v2/utils/Relation.kt
package co.instameals.dinnersolutions.api.v2.utils

import com.fasterxml.jackson.annotation.JsonValue
import java.util.UUID

class Relation<T>(val relationId: UUID) {
    var relatedObject: T? = null

    @JsonValue
    fun getRelationId(): String {
        return relationId.toString()
    }
}
